src/main/java/hudson/plugins/analysis/core/HealthAwareRecorder.java

```diff
@@ -52,7 +60,7 @@
  */
 // CHECKSTYLE:COUPLING-OFF
 @SuppressWarnings("PMD.TooManyFields")
-public abstract class HealthAwareRecorder extends Recorder implements HealthDescriptor, MatrixAggregatable, SimpleBuildStep {
+public abstract class HealthAwareRecorder extends Recorder implements HealthDescriptor, SimpleBuildStep {
```

Comment:
The problem of this change is that it makes the downstream implementations source-incompatible at least.
Binary compatibility needs to be investigated, I am not sure. There is a high probability that not all dependent plugins will be updated immediately, so it would be great to have analysis of the expected behavior in such case.